junit-vintage-engine/src/main/java/org/junit/vintage/engine/discovery/JUnit4DiscoveryRequestResolver.java

```diff
@@ -81,7 +81,23 @@ private TestClassCollector collectTestClasses(EngineDiscoveryRequest discoveryRe
 		Filter<Class<?>> classFilter = new ExclusionReasonConsumingFilter<>(adaptedFilter,
 			(testClass, reason) -> logger.fine(() -> String.format("Class %s was excluded by a class filter: %s",
 				testClass.getName(), reason.orElse("<unknown reason>"))));
-		return collector.toRequests(classFilter.toPredicate());
+
+		Predicate<Class<?>> loggingTestClass = isPotentialJunit4Predicate();
+
+		return collector.toRequests(classFilter.toPredicate().and(loggingTestClass));
+	}
+
+	private Predicate<Class<?>> isPotentialJunit4Predicate() {
+		IsPotentialJUnit4TestClass potentialTestClass = new IsPotentialJUnit4TestClass();
```

Comment:
No need to instantiate the predicate every time.

Please instantiate it once as a `static final` field as is done elsewhere in the code base for such predicates.

Also, please name the field consistent with the rest of the code base: `isPotentialJUnit4TestClass`